### PR TITLE
feat: add security headers middleware

### PR DESCRIPTION
**Pull request by hugosandelius**
Add middleware adding security-related headers.
I didn't add this middleware to the server returned from `NewHTTPServer` since it could be a breaking change.

### CODE REVIEW

**Review by alfredgunnar — APPROVED** [review]

**Comment by odsod** [comment]
@hugosandelius Nice! Two questions:

1. How big of a breaking change is it? Maybe we want to do it to make sure we have secure defaults
2. Is it different from @sharkema's change https://github.com/einride/cloudrunner-go/pull/449 where we decided to only make the change in the gateways?

**Comment by hugosandelius** [comment]
> @hugosandelius Nice! Two questions:
> 
> 1. How big of a breaking change is it? Maybe we want to do it to make sure we have secure defaults
> 2. Is it different from @sharkema's change [feat: add optional middleware for inserting security headers #449](https://github.com/einride/cloudrunner-go/pull/449) where we decided to only make the change in the gateways?

Looks like the exact same change except that was as a gRPC middleware and this is a HTTP one. Doing it in the API gateway instead makes sense to me

**Comment by odsod** [comment]
@hugosandelius But what about GraphQL gateways? There we don't have an extra gateway - maybe we should include this by default for HTTP servers but not gRPC servers?

**Comment by hugosandelius** [comment]
@odsod makes sense, changed it now to add it as a default for HTTP servers. Still a bit cautious if this can break something in some edge case, but should be relatively safe headers to add.

**Review by odsod — APPROVED** [review]
LGTM but please post in backend guild to spread awareness and see if anyone has any concerns - let's merge tomorrow afternoon?

File: cloudserver/security_headers.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+)
+
+// SecurityHeadersMiddleware adds security headers to responses.
+type SecurityHeadersMiddleware struct{}
+
+// HTTPServer provides HTTP server middleware.
+func (i *SecurityHeadersMiddleware) HTTPServer(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("X-Content-Type-Options", "nosniff")
+		w.Header().Set("Strict-Transport-Security", "max-age=31536000; includeSubDomains")
```
Comment (radhus) [inline cloudserver/security_headers.go:14]: We typically run local servers without TLS, not sure how browsers deal with HSTS on `localhost`. Did you get a chance to test that? Perhaps we could ignore this header for local servers, but not sure how we'd differentiate on that 🧐

Comment (hugosandelius) [reply cloudserver/security_headers.go:14]: `Strict-Transport-Security` header is ignored until the first time you access a URL using HTTPS, so if you only ever serve on localhost without TLS, it won't be an issue.
Seems like it becomes a problem if you ever serve any HTTPS on localhost though, so maybe worth making a special case for:
https://stackoverflow.com/questions/25277457/google-chrome-redirecting-localhost-to-https/28586593#28586593

Comment (radhus) [reply cloudserver/security_headers.go:14]: Ah, missed that it was ignored on HTTP. Then it sounds safe to me, if you ever serve HTTPS on localhost then IMO it's fair that you might have to cleanup your HSTS settings afterwards.

**Pull request merged**